browser/views.py

```diff
@@ -657,22 +656,46 @@ def add_investigator_record(request, person_id):
     template = "browser/investigator.html"
     form = InvestigatorForm()
 
+    context = {
+        'form': form,
+        'person': person,
+        'type': 'create_investigator'
+    }
+
     if request.method == 'POST':
         form = InvestigatorForm(request.POST, instance=person)
         if form.is_valid():
+            context.update({
+                'form': form
+            })
+            if form.cleaned_data.get('institution_search'):
+                if not form.cleaned_data.get('institution_id'):
+                    # Check if the checkbox is checked
+                    if form.cleaned_data.get('create_institution'):
+                        institution = Institution.objects.create(
+                            name=form.cleaned_data.get('institution_search'),
+                            changed_by=request.user
+                        )
+                        form.cleaned_data["institution_id"] = institution.id
+                    else:
+                        messages.add_message(request, messages.ERROR,
+                                             'The above institute does not exist in the database. '
+                                             'Please check the checkbox below to create '
+                                             'information about the institute.')
+                        return render(request, template, context)
```

Comment:
add the form to the context here, not in line 668 (you don't need it in the other cases, do you?).